### PR TITLE
Added github action and updated check-dups.sh

### LINKED ISSUE #92: Add scripts/check-dups.sh as GitHub action
For each PR/commit (both on main and devel branches) check if there are duplicate entries using a GitHub action.

Use [`scripts/check-dups.sh`](https://github.com/edoardottt/awesome-hacker-search-engines/blob/main/scripts/check-dups.sh). You can edit it if needed

### PR DESCRIPTION
**Pull request by milinddethe15**
fixes #92 
I have checked the action in my forked repo. It should work.

### CODE REVIEW

File: .github/workflows/check-duplicates.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Check Duplicates
+
+on:
+  pull_request:
```
Comment (edoardottt) [inline .github/workflows/check-duplicates.yml:4]: I guess you should add 
```
    branches:
      - main
```
here too, under pull_request

**Review by edoardottt — COMMENTED** [review]
- Have you tried adding a duplicated entry if it spots that correctly?
- Now the space problem is fixed? (general search engines and not working / paused)

**Comment by milinddethe15** [comment]
Yes, I tried adding a duplicated entry and it spots that correctly. 
The output of the script simply shows the duplicate links and nothing about in which catergory it is present. I wasn't able to solve the space problem.

**Comment by edoardottt** [comment]
Understood. 

- What is the purpose of `section_escaped`? It looks like it's calculated but never used... am I wrong?
- So now duplicates in those two problematic categories are spotted? Can you try 2 tests, 1 with a duplicate in `general search engine` and the other one in `not working / paused`? 
- There isn't a devel branch and I cannot test it here

**Comment by milinddethe15** [comment]
check this pr milinddethe15/awesome-hacker-search-engines#6
I have added a duplicate in `Not working / Paused` section and run it failed.

**Review by edoardottt — APPROVED** [review]
lgtm

**Pull request merged**